api/user/userModel.js

```diff
@@ -28,6 +28,13 @@ const findUserData = async (role_id, profile_id) => {
   }
 };
 
+const findUserData = async (role_id, profile_id) => {
+  return _findUserData(role_id, profile_id).then((users) => {
+    //FindUser should return only one user , so we make sure that
+    // a single user is returned instead of an array of 1 user
+    return Array.isArray(users) ? users[0] : users;
+  });
+};
 const getInbox = (okta) => {
```

Comment:
Looks great but is there a way for us to combine _findUserData and findUserData? If it gets too cluttered with combining them that's totally fine.

Reply:
Hey @sagun1151, we could combine them by using an unresolved promise like this:
`const _findUserData = async (role_id, profile_id) => {
  let user_promise = Promise.resolve([]);
  if (role_id === 1) {
    user_promise = db('profiles')
      .leftJoin(
        'super_admins',
        'profiles.profile_id',
        'super_admins.profile_id'
      )
      .where('super_admins.profile_id', profile_id);
  } else if (role_id === 2) {
    user_promise = db('profiles')
      .leftJoin('admins', 'profiles.profile_id', 'admins.profile_id')
      .where('admins.profile_id', profile_id);
  } else if (role_id === 3) {
    user_promise = db('profiles')
      .leftJoin('instructors', 'profiles.profile_id', 'instructors.profile_id')
      .where('instructors.profile_id', profile_id);
  } else if (role_id === 4) {
    user_promise = db('profiles')
      .leftJoin('parents', 'profiles.profile_id', 'parents.profile_id')
      .where('parents.profile_id', profile_id);
  } else if (role_id === 5) {
    user_promise = db('profiles')
      .leftJoin('children', 'profiles.profile_id', 'children.profile_id')
      .where('children.profile_id', profile_id);
  }
  return user_promise.then((users) => {
    //FindUser should return only one user , so we make sure that
    // a single user is returned instead of an array of 1 user
    return Array.isArray(users) ? users[0] : users;
  });
};`
But I think this way the readability is not as easy as having two functions. Thoughts? 
Let me know which way you prefer.

Reply:
Yeah, I guess it's just that last part that is the separate function and the readability does seem easier if it's 2 different functions. Maybe we can change the name to be more unique? It's whatever seems easier and understandable to you though.

Reply:
I chose the name of the function with the underscore because I was looking at the react code and it is a convention for functions that are internal and not being exported into other files to have one `_`.